src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.Statements.cs

```diff
@@ -152,6 +152,28 @@ private void EmitLoopExpression(Expression expr)
         private void EmitSwitchExpression(Expression expr, CompilationFlags flags)
         {
             SwitchExpression node = (SwitchExpression)expr;
+            
+            if (node.Cases.Count == 0)
+            {
```

Comment:
We are not helping compiler or interpreter since they will not reduce switches, but because they both have trivial for loops going over all cases, they should be able to handle 0-cases situation just fine.
As for the other consumers - we do not know if we are helping them or making things worse by making switch expression reducible, but only sometimes.

Considering that being reducible is not helping known consumers and effect on unknown consumers could go either way, let's not make switch reducible.